rollup.config.js

```diff
@@ -1,9 +1,9 @@
 import node from 'rollup-plugin-node-resolve';
 import { uglify } from 'rollup-plugin-uglify';
-import replace from 'rollup-plugin-replace';
 import typescript from 'typescript';
 import typescriptPlugin from 'rollup-plugin-typescript2';
 import filesize from 'rollup-plugin-filesize';
+import invariantPlugin from './rollup/plugin-invariant';
```

Comment:
This is a wonderfull initiative but doesn't this exclude our "exploded" CJS bundle from having benefit from this change?

Reply:
You're right about that. Maybe this argues in favor of compiling `react-apollo.esm.js` to `react-apollo.cjs.js` and pointing the `main` field of `package.json` at that CommonJS bundle?

The only immediate downside that I can think of is that one-off modules like `react-apollo/test-links` and `react-apollo/walkTree` would no longer be built if we don't run `tsc`.

Reply:
We can keep the exploded bundle by running tsc which enables the /walkTree etc, aslong as the types are exploded there should be close to no issue for web with making a cjs bundle on `main`.

Reply:
Ok, that's a good point. I think people realize that they're taking additional responsibility for their choices when they do sub-module imports like `require("react-apollo/test-links")`, so the risk of loading both the CJS bundle and some overlapping exploded modules should be limited to folks who chose to use sub-module imports.

Reply:
I think we should however include a warning about it, this way someone encountering a weird issue with it will know what to look for.

Reply:
Took a stab at this verbal warning in https://github.com/apollographql/react-apollo/pull/2771/commits/d6792eb0ba2dc883643e91fa5c31bdb7c1d466dc.